client/my-sites/stats/activity-log/index.jsx

```diff
@@ -16,14 +16,13 @@ import { first, get, groupBy, includes, isEmpty, isNull, last, range } from 'lod
  */
 import ActivityLogBanner from '../activity-log-banner';
 import ActivityLogConfirmDialog from '../activity-log-confirm-dialog';
+import ActivityLogCredentialsNotice from '../activity-log-credentials-notice';
```

Comment:
Fun tip - one thing I sometimes do when names get long is shorted them when the module itself provides additional context. In this case, the module is the `ActivityLog`, so I don't personally feel obligated to repeat that everywhere.

Feel free to do things like `import CredentialsNotice from '../activity-log-credentials-notice'` to make the code a bit less noisy.

Reply:
Oh so you're the one doing things "un-findable" Dennis? What you're saying works for this case, since the module is used only here, but back when I was coding the premium themes in Jetpack sites, I found that some modules were renamed, and then passed to another with that renaming, and inside that one it was also passed down so it wasn't trivial to find the original module and where it was coming from.
I'd personally advise against this and I'll never do it, since it makes component discovery initially confusing for others joining a project later.

Reply:
> doing things "un-findable" Dennis?

no that shouldn't be me. the names are somewhat made up on import anyway. what's fixed is the module path given to the `import` statement. that's our real key to looking up what is what.

> and then passed to another with that renaming, and inside that one it was also passed down so it wasn't trivial to find the original module and where it was coming from.

this is an incredibly painful pattern to encounter and I'm sorry you had to deal with that mess! in almost every case I encounter passing React components as props I cringe; even with simple data types I have a strong opposition to passing _any_ data _through_ something because of the detrimental effects it has on team productivity in a language like JavaScript which has such a terribly weak type system.

Redux state is there as a tool to obviate our felt needs to pass down data and functions multiple levels deep. this is the heart of the separation and isolation of concerns that Redux offers us.